sdks/python/apache_beam/io/fileio_test.py

```diff
@@ -320,6 +321,71 @@ def test_transform_on_gcs(self):
           label='Assert Checksums')
 
 
+class MatchContinuouslyTest(_TestCaseWithTempDirCleanUp):
+  def test_with_deduplication(self):
+    files = []
+    tempdir = '%s%s' % (self._new_tempdir(), os.sep)
+
+    # Create a file to be matched before pipeline
+    files.append(self._create_temp_file(dir=tempdir))
+    # Add file name that will be created mid-pipeline
+    files.append(FileSystems.join(tempdir, 'extra'))
+
+    interval = 1
```

Comment:
maybe make a shorter interval so the test will run faster? I suppose you can't run this with a TestStream? Have you tried adding a TestStream that moves time immediately?

Reply:
Yeah, I could make it shorter no problem.  Regarding `TestStream`, I am not sure how this would be implemented here, since `MatchContinuously` needs a `PBegin` as "input" (since it uses `PeriodicImpulse`)

Reply:
To add, `PeriodicImpulse` tests, use `TestPipeline` and not `TestStream`, I guess it should be the same here